export/_export.py

```diff
@@ -0,0 +1,357 @@
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import torch
+from torch import nn
+from executorch.devtools.backend_debug import get_delegation_info
+from executorch.exir.program import (
+    EdgeProgramManager,
+    ExecutorchProgramManager,
+    to_edge_transform_and_lower,
+)
+from executorch.runtime import Runtime, Verification
+from tabulate import tabulate
+from torch.ao.quantization import allow_exported_model_train_eval
+from torch.ao.quantization.quantize_pt2e import convert_pt2e, prepare_pt2e
+from torch.export import export_for_training, ExportedProgram
+
+from ._recipe import ExportRecipe
+
+
+def export(
+    model: Union[nn.Module, Dict[str, nn.Module]],
+    example_inputs: Union[List[tuple[torch.Tensor, ...]], Dict[str, List[tuple[torch.Tensor, ...]]]],
+    export_recipe: ExportRecipe,
+    name: Optional[str] = None,
+    dynamic_shapes: Optional[Union[Any, Dict[str, Any]]] = None,
+    constant_methods: Optional[Union[Dict[str, Callable]]] = None,
+    artifact_dir: Optional[str] = None,
+    apply_quantization: bool = False,
+) -> "ExportSession":
+    """
+    Create and configure an ExportSession with the given parameters.
+
+    This function provides a convenient way to create an ExportSession and
+    optionally run the export process in one step.
+
+    Args:
+        model: The PyTorch model(s) to export, either a single model or a dictionary
+              mapping method names to models
+        example_inputs: Example inputs for the model(s), either a list of input tuples
+                      or a dictionary mapping method names to lists of input tuples
+        export_recipe: Contains the configuration for the export process
+        name: Optional name for the export
+        dynamic_shapes: Optional dynamic shape specifications
+        constant_methods: Optional dictionary of constant methods
+        artifact_dir: Optional directory to store artifacts
+        apply_quantization: Whether to apply quantization during export, defaults to False
+
+    Returns:
+        A configured ExportSession instance with the export process completed if requested
+    """
+    manager = ExportSession(
+        model=model,
+        example_inputs=example_inputs,
+        export_recipe=export_recipe,
+        name=name,
+        dynamic_shapes=dynamic_shapes,
+        constant_methods=constant_methods,
+        artifact_dir=artifact_dir,
+    )
+    
+    if apply_quantization:
+        manager.export(apply_quantization=True)
+    
+    return manager
+
+
+class ExportSession:
```

Comment:
This is a good idea, I like it. But just a pet peeve, let's refactor this before starting, and organize the logic so that it can scale better, else I fear it will end up becoming a giant bowl of spaghetti.

Suggestion, make a `class Stage(ABC)` and derive others from it like `Quantize(Stage)` and `Export(Stage)` and then Session just manages the pipeline with `Stage`s. 

See - https://github.com/pytorch/executorch/blob/409447d75a1524c1acc8f8ea894c2e13dd723a79/backends/xnnpack/test/tester/tester.py#L424